contracts/strategies/PermissionedStrategy.sol

```diff
@@ -105,7 +117,14 @@ contract PermissionedStrategy is IStrategy, AccessControl {
      */
     function pullAsset(uint256 amount, address asset) public onlyWhitelisted(asset) {
         IERC20 _asset = IERC20(asset);
-        require(amount <= _asset.balanceOf(address(this)), "INSUFFICIENT FUNDS");
+        uint256 currentAssetBal = _asset.balanceOf(address(this));
+        require(amount <= currentAssetBal, "INSUFFICIENT FUNDS");
+
+        // if the asset is the vault asset, require we have enough to pay out the requested withdraws
+        if (asset == address(VAULT_ASSET)) {
+            require(currentAssetBal >= totalRequestedWithdraws, "asset needed for withdraws");
+        }
+
```

Comment:
Why is this check not required if the asset is not the vault asset? Or rather, what is supposed to happen if asset is not vault asset?